src/main/java/cubicchunks/worldgen/dependency/RegionDependency.java

```diff
@@ -26,75 +26,73 @@
 
 import java.util.Collection;
 import java.util.HashSet;
-import java.util.Set;
 
-import cubicchunks.util.AddressTools;
 import cubicchunks.util.CubeCoords;
 import cubicchunks.world.cube.Cube;
 import cubicchunks.worldgen.GeneratorStage;
 
 public class RegionDependency implements Dependency {
 	
 	private GeneratorStage targetStage;
-	private Set<Requirement> requirements;
-	
-	public RegionDependency(Cube cube, GeneratorStage targetStage, int radius) {
+
+	private int xLow;
+	private int xHigh;
+	private int yLow;
+	private int yHigh;
+	private int zLow;
+	private int zHigh;
+
+
+	public RegionDependency(GeneratorStage targetStage, int radius) {
 
 		this.targetStage = targetStage;
-		this.requirements = new HashSet<Requirement>();
 
-		int cubeX = cube.getX();
-		int cubeY = cube.getY();
-		int cubeZ = cube.getZ();
-		
-		int xLow = -radius;
-		int xHigh = radius;
-		int yLow = -radius;
-		int yHigh = radius;
-		int zLow = -radius;
-		int zHigh = radius;
-
-		for (int x = xLow; x <= xHigh; ++x) {
-			for (int y = yLow; y <= yHigh; ++y) {
-				for (int z = zLow; z <= zHigh; ++z) {
-					if (x != 0 || y != 0 || z != 0) {
-						CubeCoords coords = new CubeCoords(cubeX + x, cubeY + y, cubeZ + z);
-						requirements.add(new Requirement(coords, targetStage));
-					}
-				}
-			}
-		}
+		this.xLow = -radius;
+		this.xHigh = radius;
+		this.yLow = -radius;
+		this.yHigh = radius;
+		this.zLow = -radius;
+		this.zHigh = radius;
 	}
 	
-	public RegionDependency(Cube cube, GeneratorStage stage, int xLow, int xHigh, int yLow, int yHigh, int zLow, int zHigh) {
+	public RegionDependency(GeneratorStage targetStage, int xLow, int xHigh, int yLow, int yHigh, int zLow, int zHigh) {
```

Comment:
I would suggest using `Vec3i`s instead of ints here, argument order isn't obvious from code.